File: VB-Examples/02_FindAndReplace/FindKeyWordsInParagraph.vb
Imports Spire.Doc
Imports Spire.Doc.Documents

Namespace FindKeyWordsInParagraph
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			'Input file path
			Dim input As String = "..\..\..\..\..\..\Data\Sample.docx"

			'Output file path
			Dim output As String = "FindKeyWordsInParagraph_output.docx"

			'Create word document
			Dim document As New Document()

			'Load a document
			document.LoadFromFile(input)

			'Get the first section
			Dim s As Section = document.Sections(0)

			'Get the second paragraph
			Dim para As Paragraph = s.Paragraphs(1)

			'Find all matched keywords
			Dim textSelections() As TextSelection = para.FindAllString("Word", False, True)

			'Highlight text
			For Each selection As TextSelection In textSelections
				selection.GetAsOneRange().CharacterFormat.HighlightColor= Color.FromArgb(255, 255, 0)
			Next selection

			' Save to file
			document.SaveToFile(output, FileFormat.Docx2019)

			'Dispose the document
			document.Dispose()

			'Launching the Word file.
			WordDocViewer(output)
		End Sub

		Private Sub WordDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

	End Class
End Namespace

File: CS-Examples/02_FindAndReplace/FindKeyWordsInParagraph.cs
using System;
using System.Drawing;
using System.Windows.Forms;
using Spire.Doc;
using Spire.Doc.Documents;

namespace FindKeyWordsInParagraph
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            //Input file path
            String input = "..\\..\\..\\..\\..\\..\\Data\\Sample.docx";

            //Output file path
            String output = "FindKeyWordsInParagraph_output.docx";

            //Create word document
            Document document = new Document();

            //Load a document
            document.LoadFromFile(input);

            //Get the first section
            Section s = document.Sections[0];

            //Get the second paragraph
            Paragraph para = s.Paragraphs[1];

            //Find all matched keywords
            TextSelection[] textSelections = para.FindAllString("Word", false, true);

            //Highlight text
            foreach (TextSelection selection in textSelections)
            {
                selection.GetAsOneRange().CharacterFormat.HighlightColor= Color.FromArgb(255, 255, 0);
            }

            // Save to file
            document.SaveToFile(output, FileFormat.Docx2019);

            //Dispose the document
            document.Dispose();

            //Launching the Word file.
            WordDocViewer(output);
        }

        private void WordDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

    }
}
